aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1602,6 +1602,110 @@ Tensor& linalg_norm_out(Tensor& result, const Tensor& self, std::string ord, opt
   return linalg_norm_out_impl(result, self, c10::nullopt, ord, opt_dim, keepdim, opt_dtype);
 }
 
+// Numerical or None norms
+Tensor linalg_cond(const Tensor& self, optional<Scalar> opt_ord) {
+  TORCH_CHECK(self.numel() > 0, "linalg_cond is not defined for empty tensors.");
+  TORCH_CHECK(self.dim() >= 2, "Tensor of matrices must have at least 2 dimensions.");
+
+  Tensor self_inverse, result;
+
+  // The default case is using 2-norm
+  Scalar ord = opt_ord.has_value() ? opt_ord.value() : 2;
+
+  // If ord == None or ord == ±2
+  if (std::abs(ord.toDouble()) == 2.0) {
+    auto singular_values = std::get<1>(at::svd(self));
+    auto s_max = std::get<0>(singular_values.max(/*dim=*/-1));
+    auto s_min = std::get<0>(singular_values.min(/*dim=*/-1));
+    if (ord.toDouble() == -2.0) {
+      result = s_min / s_max;
+    }
+    else {
+      result = s_max / s_min;
+    }
+    // a trick to convert FLT_MAX or DBL_MAX to INFINITY
+    // result might also contains infs already then inf + inf - inf = nan
+    // this will get replaced by inf
+    result = result + result - result;
+    result = at::nan_to_num(result, INFINITY);
+    return result;
+  }
+  // ord == ±1 ord == ±inf
+  else {
```

Comment:
I don't think you actually need the else branch here because the previous branch always returns